Project1/id_cost_map.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <map>
+#include "ID.h"
+#include "logic_pair.h"
+#include "GridCell.h"
+class id_cost_map : private std::map<ID, CostID>
```

Comment:
why private inheritance if you expose all the methods?

Reply:
I was not going to do them all, but caved in is why